floors/floors.go

```diff
@@ -0,0 +1,124 @@
+package floors
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/currency"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	DEFAULT_DELIMITER      string = "|"
+	CATCH_ALL              string = "*"
+	SKIP_RATE_MIN          int    = 0
+	SKIP_RATE_MAX          int    = 100
+	MODEL_WEIGHT_MAX_VALUE int    = 100
+	MODEL_WEIGHT_MIN_VALUE int    = 0
+	ENFORCE_RATE_MIN       int    = 0
+	ENFORCE_RATE_MAX       int    = 100
+)
+
+type Config struct {
+	Enabled           bool
+	EnforceRate       int
+	EnforceDealFloors bool
+}
+
+func NewConfig(priceFloor config.PriceFloors) *Config {
+
+	config := Config{
+		Enabled:           priceFloor.Enabled,
+		EnforceRate:       priceFloor.EnforceFloorsRate,
+		EnforceDealFloors: priceFloor.EnforceDealFloors,
+	}
+
+	return &config
+}
+
+func (fc *Config) IsFloorEnabled() bool {
+	return fc.Enabled
+}
+
+func (fc *Config) GetEnforceRate() int {
+	return fc.EnforceRate
+}
+
+func (fc *Config) EnforceDealFloor() bool {
+	return fc.EnforceDealFloors
+}
+
+type Floor interface {
+	IsFloorEnabled() bool
+	GetEnforceRate() int
+	EnforceDealFloor() bool
+}
+
+// ModifyImpsWithFloors will validate floor rules, based on request and rules prepares various combinations
+// to match with floor rules and selects appripariate floor rule and update imp.bidfloor and imp.bidfloorcur
+func ModifyImpsWithFloors(floorExt *openrtb_ext.PriceFloorRules, request *openrtb2.BidRequest, conversions currency.Conversions) []error {
+	var (
+		floorErrList      []error
+		floorModelErrList []error
+		floorVal          float64
+	)
+	floorData := floorExt.Data
+	if floorData == nil {
+		return nil
+	}
+
+	floorModelErrList = validateFloorSkipRates(floorExt)
+	if len(floorModelErrList) > 0 {
+		return floorModelErrList
+	}
+
+	floorData.ModelGroups, floorModelErrList = validateFloorModelGroups(floorData.ModelGroups)
+	if len(floorData.ModelGroups) == 0 {
+		return floorModelErrList
+	} else if len(floorData.ModelGroups) > 1 {
+		floorData.ModelGroups = selectFloorModelGroup(floorData.ModelGroups, rand.Intn)
+	}
+
+	if floorData.ModelGroups[0].Schema.Delimiter == "" {
```

Comment:
Consider assigning a model group variable, since there is only one left at this point in the code. Something like:
```
modelGroup := floorData.ModelGroups[0]
```